### PR TITLE
Reorganize pipelines and update CodeQL

### PR DESCRIPTION
**Pull request by jander-msft**
###### Summary

- Move pipelines (except for `dotnet-monitor.yml`), jobs, and stages under `/eng/pipelines` for consistency with other .NET repos and organization within the repo.
- Refactor build platform and configuration matrix to `eng/pipelines/jobs/platform-matrix.yml` pipeline template so that it can be reused among multiple pipelines.
- Update `/eng/pipelines/jobs/build-test.yml` to not run test for ARM64 builds and write this skip out as a build step.
- Add `/eng/pipelines/jobs/bulid-codeql.yml` and `/eng/pipelines/dotnet-monitor-codeql.yml` to run CodeQL on all release configurations of the build platforms and architectures.

Example product build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2067059&view=results
Example CodeQL build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2067060&view=results

### CODE REVIEW

File: eng/pipelines/jobs/build-codeql.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Builds dotnet-monitor for a specific platform and configuratoin with CodeQL analysis enabled
```
Comment (kkeirstead) [inline eng/pipelines/jobs/build-codeql.yml:1]:
```suggestion
# Builds dotnet-monitor for a specific platform and configuration with CodeQL analysis enabled
```

File: eng/pipelines/stages/preparerelease.yml [hunk 2]
```diff
@@ -1,84 +1,84 @@
...
+        name: NetCore1ESPool-Internal
+        demands: ImageOverride -equals 1es-windows-2019
+    variables:
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), startsWith(variables['Build.SourceBranch'], 'refs/heads/internal/release/'), startsWith(variables['Build.SourceBranch'], 'refs/heads/test/release/'))) }}:
+      - group: DotNet-Diagnostics-Storage
+      - group: DotNet-DotNetStage-Storage
+      - group: Release-Pipeline
+    steps:
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core runtime 3.1.x'
```
Comment (kkeirstead) [inline eng/pipelines/stages/preparerelease.yml:24]: Do we still want this?

Comment (jander-msft) [reply eng/pipelines/stages/preparerelease.yml:24]: I didn't want to change functionality; I can look into this separately.

File: eng/pipelines/jobs/build.yml [hunk 3]
```diff
@@ -88,7 +100,7 @@ jobs:
       - ${{ variable }}
 
     # Component Governance does not work on Musl
```
Comment (wiktork) [inline eng/pipelines/jobs/build.yml:102]: Is this still true?

File: eng/pipelines/jobs/platform-matrix.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
+# Executes a specified job on each supported platform and configuration
+
+parameters:
+  # Job template to run on applicable platforms
+  jobTemplate: ''
```
Comment (wiktork) [inline eng/pipelines/jobs/platform-matrix.yml:5]: Any notable changes in the matrix?

**Review by wiktork — APPROVED** [review]

File: eng/pipelines/jobs/platform-matrix.yml (hunk 4 above)
Comment (jander-msft) [reply eng/pipelines/jobs/platform-matrix.yml:5]: Nothing changed in the matrix and all of the effective parameters are the same.

**Pull request merged**